src/vm/Components/Assembler/Assembler.java

```diff
@@ -0,0 +1,208 @@
+package vm.Components.Assembler;
+
+import java.io.FileNotFoundException;
+import java.util.HashMap;
+import java.util.Map;
+
+import vm.Components.FileHandlers.Reader;
+import vm.Components.FileHandlers.Writer;
+
+public class Assembler {
+    private int addressCounter;
+    private int lineCounter;
+    private Map<String, InstructionsWrapper> instructionsMap;
+    private Map<String, Integer> symbolTable;
+
+    public Assembler() {
+        addressCounter = 0;
+        lineCounter = 1;
+        instructionsMap = new HashMap<>();
+        symbolTable = new HashMap<>();
+        setInstructionData();
+        assemble();
+    }
+
+    public void assemble() {
+        try {
+            firstPass();
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
+            System.out.println("Erro no processo do primeiro passo do montador!");
+        } catch (AssemblerException e) {
+            return;
+        } finally {
+            System.out.println(symbolTable);
+        }
+
+        try {
+            secondPass();
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
+            System.out.println("Erro no processo do segundo passo do montador!");
+        }
+    }
+
+    private void firstPass() throws FileNotFoundException, AssemblerException {
+        Reader reader = new Reader("MASMAPRG.ASM");
+        String line = reader.readLine();
+
+        // retira comentarios
+        line = line.split(";")[0];
+
+        // endereco sera zero enquanto desconhecido
+        int address = 0;
+        while (line != null) {
+
+            if (line.length() > 80) {
+                throw new AssemblerException("Assembler Exception: Invalid input at line " + lineCounter + ". Lines should have a maximum length of 80 characters.");
+            }
+
+            String [] params = line.split(" ");
+            if (params.length > 4) {
```

Comment:
acho que esse 80 e esse 4 poderiam ser atributos finais na classe, pra ficar mais claro o que indicam. o que achas?
tipo:
```
public static final LINE_MAX_PARAMS = 4

...

if(params.length < LINE_MAX_PARAMS) { ... } `
```

Reply:
resolved in commit https://github.com/guivahl/programacao-de-sistemas/commit/7fb99ebb5fb76f7c06d91c084deb1fb094fa2210